src/main/scala/org/apache/mesos/chronos/scheduler/jobs/JobStats.scala

```diff
@@ -80,6 +80,136 @@ class JobStats @Inject()(clusterBuilder: Option[Cluster.Builder], config: Cassan
     jobStates.remove(job.name)
   }
 
+  /**
+   * Queries Cassandra table for past and current job statistics by jobName
+   * and limits by numTasks. The result is not sorted by execution time
+   * @param jobName
+   * @param numTasks
+   * @return list of cassandra rows
+   */
+  private def getJobStatsByName(jobName: String, numTasks: Int): Option[List[Row]] = {
+    var rowsListFinal: Option[List[Row]] = None
+    try {
+      getSession match {
+        case Some(session: Session) =>
+          val query = s"SELECT * FROM ${config.cassandraTable()} WHERE ${JOB_NAME}='${jobName}';"
+          val prepared = statements.getOrElseUpdate(query, {
+            session.prepare(
+              new SimpleStatement(query)
+                .setConsistencyLevel(readConsistencyLevel())
+                .asInstanceOf[RegularStatement]
+            )
+          })
+
+          val resultSet = session.execute(prepared.bind())
+          val rowsList = resultSet.all().asScala.toList
+          rowsListFinal = Some(rowsList)
+        case None => rowsListFinal = None
+      }
+    } catch {
+      case e: NoHostAvailableException =>
+        resetSession()
+        log.log(Level.WARNING, "No hosts were available, will retry next time.", e)
+      case e: QueryExecutionException =>
+        log.log(Level.WARNING,"Query execution failed:", e)
+      case e: QueryValidationException =>
+        log.log(Level.WARNING,"Query validation failed:", e)
+    }
+    rowsListFinal
+  }
+
+  /**
+   * Compare function for TaskStat by most recent date.
+   */
+  private def recentDateCompareFnc(a: TaskStat, b: TaskStat): Boolean = {
+    var compareAscDate = a.taskStartTs match {
+      case Some(aTs: DateTime) => {
+        b.taskStartTs  match {
+          case Some(bTs: DateTime) => {
+            aTs.compareTo(bTs) <= 0
+          }
+          case None => {
+            false
+          }
+        }
+      }
+      case None => {
+        true
+      }
+    }
+    !compareAscDate
+  }
+
+  /**
+   * Returns most recent tasks by jobName and returns only numTasks
+   * @param jobName
+   * @param numTasks
+   * @return returns a list of past and currently running tasks,
+   *         the first element is the most recent.
+   */
+  def getParsedJobStatsByName(jobName: String, numTasks: Int): List[TaskStat] = {
+    val taskMap = Map[String, TaskStat]()
```

Comment:
The reason I did not go down the route of persisting this map is that now there will be two sources of information to update.  Every time an insertion is done to the existing cassandra table (i.e. chronos table), another update will have to be done to this aggregate taskMap data in cassandra.  I was worried more bugs may result from that.  Hence, I opted to have one source of truth, which is the existing data storage setup.

In regards to having incomplete data (no start date and/or no end date), I can only see a situation in which there is a start date but no end date (i.e. task still running).  If the reverse happens, I would argue that the bug happens further up the chain.

Thoughts?